Libraries/LibWeb/CSS/CSSFontFaceRule.cpp

```diff
@@ -58,83 +58,102 @@ String CSSFontFaceRule::serialized() const
     builder.append("@font-face { "sv);
 
-    // 2. The string "font-family:", followed by a single SPACE (U+0020).
-    builder.append("font-family: "sv);
-
-    // 3. The result of performing serialize a string on the rule’s font family name.
-    builder.append(descriptors.descriptor(DescriptorID::FontFamily)->to_string(SerializationMode::Normal));
-
-    // 4. The string ";", i.e., SEMICOLON (U+003B).
-    builder.append(';');
+    // 2. If the rule’s associated font-family is present, follow these substeps:
+    //    The string "font-family:", followed by a single SPACE (U+0020).
+    //    followed by the result of performing serialize a string on the rule’s font family name.
+    //    followed by the string ";", i.e., SEMICOLON (U+003B).
+    if (auto font_family = descriptors.descriptor(DescriptorID::FontFamily)) {
+        builder.append("font-family: "sv);
+        builder.append(descriptors.descriptor(DescriptorID::FontFamily)->to_string(SerializationMode::Normal));
+        builder.append(';');
+        is_empty = false;
+    }
 
-    // 5. If the rule’s associated source list is not empty, follow these substeps:
+    // 3. If the rule’s associated source list is not empty, a single SPACE (U+0020),
+    //    followed by the string "src:", followed by a single SPACE (U+0020).
+    //    followed by the result of invoking serialize a comma-separated list on performing serialize a URL or serialize a LOCAL for each source on the source list.
+    //    followed by the string ";", i.e., SEMICOLON (U+003B).
     if (auto sources = descriptors.descriptor(DescriptorID::Src)) {
-        // 1. A single SPACE (U+0020), followed by the string "src:", followed by a single SPACE (U+0020).
-        builder.append(" src: "sv);
-
-        // 2. The result of invoking serialize a comma-separated list on performing serialize a URL or serialize a LOCAL for each source on the source list.
+        if (!is_empty)
+            builder.append(' ');
+        builder.append("src: "sv);
         builder.append(sources->to_string(SerializationMode::Normal));
-
-        // 3. The string ";", i.e., SEMICOLON (U+003B).
         builder.append(';');
+        is_empty = false;
     }
 
-    // 6. If rule’s associated unicode-range descriptor is present, a single SPACE (U+0020), followed by the string "unicode-range:", followed by a single SPACE (U+0020), followed by the result of performing serialize a <'unicode-range'>, followed by the string ";", i.e., SEMICOLON (U+003B).
+    // 4. If rule’s associated unicode-range descriptor is present, a single SPACE (U+0020), followed by the string "unicode-range:", followed by a single SPACE (U+0020), followed by the result of performing serialize a <'unicode-range'>, followed by the string ";", i.e., SEMICOLON (U+003B).
     if (auto unicode_range = descriptors.descriptor(DescriptorID::UnicodeRange)) {
+        if (!is_empty)
+            builder.append(' ');
         builder.append(" unicode-range: "sv);
```

Comment:
The leading space is no longer necessary here. The same is true for each descriptor below that follows this pattern.

Reply:
Hi @Psychpsyo thanks for clarification, I've reverted the comments to be aligned with the spec.